### PR TITLE
Art/refactor some token functions

### PR DESCRIPTION
**Pull request by whenceforth**
I'd like to minimize messing with internals of submodule state in Main, especially stuff like peering into Dicts: 
```
Dict.get tokenAddress tokenState.tokenBalances
```

so here I introduce some stuff to move in that direction for Token.elm.

### CODE REVIEW

File: src/elm/Eth/Token.elm [hunk 1]
```diff
@@ -271,6 +267,16 @@ giveTokenInformation wrapper =
 -- FUNCTIONS
 
 
+maybeElement : Maybe a -> (a -> Html msg) -> Html msg
```
Comment (hayesgm) [inline src/elm/Eth/Token.elm]: Can you reverse the order of the parameters here? That way we can use this like:

```elm
myValue
|> maybeElement \x -> text "x"
```

Generally, the most "privileged" argument is the last element in elm due to the nature of `|>`

File: src/elm/Eth/Token.elm [hunk 2]
```diff
@@ -198,32 +204,22 @@ tokenIsApproved token tokenState =
...
+                            getCustomerTokenDetails token tokenState
 
                         tokenBalanceEl =
-                            case Dict.get tokenAddress tokenBalances of
-                                Nothing ->
-                                    text ""
-
-                                Just balance ->
-                                    text <| Decimal.toString balance
+                            maybeElement balanceAndAllowance.balance (\balance -> text <| Decimal.toString balance)
```
Comment (hayesgm) [inline src/elm/Eth/Token.elm]: Not that it matters much, but I think this is one of those times that you can use <<, as in:

`maybeElement balanceAndAllowance.balance (text << Decimal.toString balance)`

File: src/elm/Eth/Token.elm (hunk 1 above)
Comment (whenceforth) [reply src/elm/Eth/Token.elm]: done

File: src/elm/Eth/Token.elm (hunk 2 above)
Comment (whenceforth) [reply src/elm/Eth/Token.elm]: hmm.. not having a lot of luck with that.

**Pull request merged**